.github/workflows/unit-tests.yaml

```
@@ -0,0 +1,41 @@
+name: "unit-tests"
+
+on:
+  pull_request:
+  push:
+  schedule:
+    # Run daily at 00:20 UTC
+    - cron: "20 0 * * *"
+  workflow_dispatch:
+
+permissions: {}
+
+concurrency:
+  group: "${{github.workflow}}-${{github.ref}}"
+
+  # Cancel in-progress jobs for efficiency
+  cancel-in-progress: true
+
+jobs:
+  unit-tests:
+    strategy:
+      matrix:
+        os:
```

```suggestion
        os:
          - "macos-14"
          - "macos-14"
```

Unless there are issues with running on macOS? The reason to test on macOS is that some developers use macOS.

---

There are issues with running on MacOS. The boost test cases have a dependency on /dev/true which is only available on some linux distributions.

---

Ah gotcha. Can you add that to the PR description and link tot he issue you opened? That way if someone else comes back to this PR/change later, they know why we didn't add macOS.